### ISSUE TITLE
Hyperlink Cells Randomly Changing to String Cells

### ISSUE THREAD

**Issue by FriendOfTheNight** [exceljs/exceljs#684]
Hi,

I'm experiencing a pretty nasty bug where hyperlink cells in rows I have added change to plain string cells after adding a new row. Most of the time adding a new row does not cause the issue but it happens maybe 1/10th of the time. It only seems to affect the cells in the last row (the last row before the new row was added).

Here is how I'm adding rows:

```
  let book = [];
  let isbn13 = '';
  for (let i = 1; i <= worksheet.columnCount; i++) {
    let cell = $('#add-row-display tbody td:nth-child('+i+')')[0];
    if (i === 1) {
      isbn13 = cell.innerText;
    }
    if (i === 3) {
      let obj = {
        isHyperlink: true,
        text: cell.innerText,
        hyperlink: 'https://www.amazon.com/dp/' + convISBN13toISBN10(isbn13)
      };
      book.push(obj);
    }
    else {
      book.push(cell.innerText);
    }
  }
  worksheet.addRow(book);
````

As I cannot reproduce the issue consistently I'm hoping someone can point me in the right direction. I have debugged the code extensively already but can't find what's causing this to happen. Before adding a new row the relevant cell in the previous row has a ValueType of 5, isHyperlink is true, and the hyperlink is stored in the hyperlink property. After adding a new row the same cell will have a ValueType of 3, isHyperlink is false, and the hyperlink is gone.

Any help with this would be greatly appreciated.

Thanks,
Brian

**Comment by FriendOfTheNight** [comment]
Forgot to mention I'm using version 1.6.2.

**Comment by FriendOfTheNight** [comment]
Update:

Switching to version 1.6.0 seemed to fix the issue.

When I tried 1.6.1 the hyperlinks stayed in my application but after saving the worksheet a handful of them were switched back to strings.

**Comment by LesterLyu** [comment]
Check: https://github.com/guyonroche/exceljs/pull/675

**Comment by Siemienik** [comment]
Hello @FriendOfTheNight, did You resolve it or has any additional information?

**Comment by FriendOfTheNight** [comment]
Hi @Siemienik,

The issue went away as I was fixing and working around other bugs I encountered so I can't say for sure what fixed it. The application I built with it is using 1.6.0 now and starts with a template file where the last few rows never change so I think that might have been what fixed it. I also had to implement a sort by column function where I had to loop through the rows and add them to a new worksheet copy to workaround other issues so that could have been what fixed it too. Eventually I removed the isHyperlink property from the object but pretty sure one of my other workarounds had fixed it by then.

-Brian
